tests/system/contracts/BridgeOut.sol

```diff
@@ -0,0 +1,151 @@
+
+        emit Transfer(address(0), to, amount);
+    }
+
+    // Approve function - allows spender to use owner's tokens
+    function approve(address spender, uint256 amount) public returns (bool) {
+        require(spender != address(0), "ERC20: approve to zero address");
+
+        _allowances[msg.sender][spender] = amount;
+
+        emit Approval(msg.sender, spender, amount);
+        return true;
+    }
+
+    // TransferFrom function - transfers tokens on behalf of another address
+    function transferFrom(address from, address to, uint256 amount) public returns (bool) {
+        require(from != address(0), "ERC20: transfer from zero address");
+        require(to != address(0), "ERC20: transfer to zero address");
+        require(_balances[from] >= amount, "ERC20: transfer amount exceeds balance");
+        require(_allowances[from][msg.sender] >= amount, "ERC20: transfer amount exceeds allowance");
+
+        _balances[from] -= amount;
+        _balances[to] += amount;
+        _allowances[from][msg.sender] -= amount;
+
+        emit Transfer(from, to, amount);
+        return true;
+    }
+
+    function burnFrom(address from, uint256 amount) public {
+        require(from != address(0), "ERC20: burn from zero address");
+        require(_balances[from] >= amount, "ERC20: burn amount exceeds balance");
+        require(_allowances[from][msg.sender] >= amount, "ERC20: burn amount exceeds allowance");
+
+        _balances[from] -= amount;
+        _totalSupply -= amount;
+        _allowances[from][msg.sender] -= amount;
+
+        emit Transfer(from, address(0), amount);
+    }
+}
+
+/// @title BridgeOut
+/// @notice Handles various bridgeOut scenarios for Mezo native asset bridge.
+contract BridgeOut {
+    // AssetsBridge precompile address on Mezo.
+    address private constant bridgePrecompile = 0x7B7C000000000000000000000000000000000012;
+    // BTC precompile address on Mezo
+    address private constant btcPrecompile = 0x7b7C000000000000000000000000000000000000;
+
+    function bridgeOutBTCToBitcoinSuccess(bytes calldata recipient, uint256 amount) external payable {
+	bool okApprove = IBTC(btcPrecompile).approve(bridgePrecompile, amount);
+        require(okApprove, "couldn't approve bridge for transferFrom");
+
+	bool okBridgeOut = IAssetsBridge(bridgePrecompile).bridgeOut(btcPrecompile, amount, 1, recipient);
+        require(okBridgeOut, "couldn't bridge out btc");
+    }
+
+    function bridgeOutBTCToBitcoinReverts(bytes calldata recipient, uint256 amount) external payable {
+	bool okApprove = IBTC(btcPrecompile).approve(bridgePrecompile, amount);
+        require(okApprove, "couldn't approve bridge for transferFrom");
+
+	bool okBridgeOut = IAssetsBridge(bridgePrecompile).bridgeOut(btcPrecompile, amount, 1, recipient);
+        require(okBridgeOut, "couldn't bridge out btc");
+
+	// now just revert
+	revert("revert triggered");
+    }
+
+    function bridgeOutBTCToEthereum(address recipient) external payable {
```

Comment:
For info, from here, they were just placeholder from when I was starting to think of what to test. However this is all covered by the EOA tests, and I'm not sure it makes sense to implement all these.

Reply:
See https://github.com/mezo-org/mezod/pull/523#discussion_r2260137075